### PR TITLE
Allow greater values in FIXED_POINT HF adjustment

### PR DESCRIPTION
**Pull request by eustas**
This adresses 2 UBs

### CODE REVIEW

**Comment by eustas** [comment]
Move to draft, please.

**Comment by eustas** [comment]
Ready for review

File: libfaad/sbr_hfadj.c [hunk 1]
```diff
@@ -301,31 +301,24 @@ static real_t find_log2_E(sbr_info *sbr, uint8_t k, uint8_t l, uint8_t ch)
         E >>= amp1;
         if (e < 0 || e >= 64 || E < 0 || E > 24)
             return LOG2_MIN_INF;
+        E -= 12;
 
-        if (ch == 0)
+        if (ch != 0)  // L/R anti-symmetry
```
Comment (fabiangreffrath) [inline libfaad/sbr_hfadj.c:306]: He?

**Review by fabiangreffrath — APPROVED** [review]
Sorry, but it's hard to follow these changes. I trust you that it's good, but would you mind elaborating a bit? What's the core part of the change set and how about the dozens of lines that you just dropped at the end?

**Comment by eustas** [comment]
Sure, will add more clarification notes. I can even can split this into parts:
 - 64-bit helper functions / intermediate values
 - find_log2_(E|Q) refactoring; it does not affect calculations, but makes it more clear what values are addressed
 - merging reglar/fixed-point codepaths
 - boost capping

Will do that tomorrow morning.

File: libfaad/sbr_hfadj.c (hunk 1 above)
Comment (eustas) [reply libfaad/sbr_hfadj.c:306]: It is a comment that left (`ch == 0`) and right (`ch != 0`) channels have joint E/Q, which means the bigger result acquires one of them, the smaller gets the other one (anti-symmetry). This change removes code repetition - easier to read and understand.
